hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -1081,6 +1082,18 @@ object MergeIntoHoodieTableCommand {
         parameters.getOrElse(config.key, config.defaultValue().toString).toBoolean
     }.getOrElse(false)
   }
+
+  def useCustomMergeMode(parameters: Map[String, String]): Boolean = {
+    val inferredMergeConfigs = HoodieTableConfig.inferCorrectMergingBehavior(
+      RecordMergeMode.getValue(parameters.getOrElse(DataSourceWriteOptions.RECORD_MERGE_MODE.key(), null)),
+      parameters.getOrElse(DataSourceWriteOptions.PAYLOAD_CLASS_NAME.key(), ""),
+      parameters.getOrElse(DataSourceWriteOptions.RECORD_MERGE_STRATEGY_ID.key(), ""),
+      parameters.getOrElse(PRECOMBINE_FIELD.key(), null),
+      HoodieTableVersion.fromVersionCode(parameters.getOrElse(
+        HoodieWriteConfig.WRITE_TABLE_VERSION.key,
+        HoodieTableVersion.current().versionCode().toString).toInt))
+    inferredMergeConfigs.getLeft.equals(RecordMergeMode.CUSTOM)
```

Comment:
No need to infer the merge mode as the merge mode table config is always available in table version 8.